File: backend/src/main/java/com/khojdu/backend/entity/Review.java
package com.khojdu.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "reviews")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "property_id", nullable = false)
    private Property property;

    @ManyToOne
    @JoinColumn(name = "tenant_id", nullable = false)
    private User tenant;

    @ManyToOne
    @JoinColumn(name = "landlord_id", nullable = false)
    private User landlord;

    // Ratings (1-5 scale)
    @Column(name = "overall_rating")
    private Integer overallRating;

    @Column(name = "cleanliness_rating")
    private Integer cleanlinessRating;

    @Column(name = "location_rating")
    private Integer locationRating;

    @Column(name = "value_rating")
    private Integer valueRating;

    @Column(name = "landlord_rating")
    private Integer landlordRating;

    @Column(name = "review_text")
    private String reviewText;

    private String pros;
    private String cons;

    // Verification
    @Column(name = "is_verified")
    private Boolean isVerified = false;

    @Column(name = "stay_duration_months")
    private Integer stayDurationMonths;

    @CreatedDate
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/types/review.ts
export interface ReviewSummary {
  averageRating: number
  totalReviews: number
  ratingDistribution: Record<number, number>
  cleanlinessAverage?: number
  locationAverage?: number
  valueAverage?: number
  landlordAverage?: number
}

export interface Review {
  id: string
  propertyId: string
  propertyTitle?: string
  tenantId?: string
  tenantName?: string
  tenantProfileImage?: string
  overallRating: number
  cleanlinessRating?: number
  locationRating?: number
  valueRating?: number
  landlordRating?: number
  reviewText?: string
  pros?: string
  cons?: string
  isVerified?: boolean
  stayDurationMonths?: number
  createdAt?: string
  updatedAt?: string
}
